board.cpp

```diff
@@ -0,0 +1,190 @@
+#include "board.h"
+
+const std::vector<CreateFn> Board::pieceNames = { &Create<Square>, &Create<TBlock>, &Create<LBlockL>, &Create<LBlockR>, &Create<Straight>, &Create<ZBlock>, &Create<SBlock>};
+
+Board::Board() {
+	this->board = std::vector<int>(BOARD::BOARD_SIZE, 0);
+	this->blocksPerRow = std::vector<int>(BOARD::COLUMN_SIZE, 0);
+	this->currentHeight = 0;
+	this->currentPiece = new Piece();
+	this->absPiecePositionCol = 0;
+	this->absPiecePositionRow = 0;
+	this->rotation = 0;
+}
+
+int& Board::operator[] (int rowCol){
+	return board[rowCol];
+}
+
+
+int& Board::operator[] (indices rowCol) {
+	return board[rowCol.i * 10 + rowCol.j];
+}
+
+inline int Board::getPieceCol(int i) {
+	return this->absPiecePositionCol + (*(this->currentPiece->relXPositions))[rotation * 4 + i];
+}
+
+inline int Board::getPieceRow(int i) {
+	return this->absPiecePositionRow + (*(this->currentPiece->relYPositions))[rotation * 4 + i];
+}
+
+inline int Board::getPieceRowCol(int i) {
+	return 10 * this->absPiecePositionRow + this->absPiecePositionCol + (*(this->currentPiece->relXPositions))[rotation * 4 + i] + 10 * (*(this->currentPiece->relYPositions))[rotation * 4 + i];
+}
+
+bool Board::checkPieceOverlaps(int offsetX, int offsetY) {
```

Comment:
I refuse to read any of the bound checking logic until you go to a 2D board to simplify or prove to me via metrics that having a 1D board is better from a performance standpoint. Otherwise, I think it just leads to harder to read/debug code.